Tests/GraphitiTests/ProductAPI/ProductResolver.swift

```diff
@@ -0,0 +1,89 @@
+import Foundation
+import Graphiti
+import NIO
+
+struct ProductResolver {
+    var sdl: String
+
+    struct ProductArguments: Codable {
+        let id: String
+    }
+
+    func product(context: ProductContext, arguments: ProductArguments) -> Product? {
+        context.getProduct(id: arguments.id)
+    }
+
+    struct DeprecatedProductArguments: Codable {
+        let sku: String
+        let package: String
+    }
+
+    func deprecatedProduct(context: ProductContext, arguments: DeprecatedProductArguments) -> DeprecatedProduct? {
+        context.getDeprecatedProduct(sku: arguments.sku, package: arguments.package)
+    }
+}
+
+extension ProductResolver: FederationResolver {
+    static let encoder = JSONEncoder()
+    static let decoder = JSONDecoder()
+    static let entityKeys: [(entity: FederationEntity.Type, keys: [FederationEntityKey.Type])] = [
+        (Product.self, [Product.EntityKey1.self, Product.EntityKey2.self, Product.EntityKey3.self]),
+        (DeprecatedProduct.self, [DeprecatedProduct.EntityKey.self]),
+        (ProductResearch.self, [ProductResearch.EntityKey.self]),
+        (ProductUser.self, [ProductUser.EntityKey.self]),
+    ]
```

Comment:
We map up our Entities to Keys through the resolver already - is there any way we could reuse that definition instead of requiring it twice?